### PR TITLE
REF: move Loc method that belongs on Index

### PR DESCRIPTION
**Pull request by jbrockmendel**

### CODE REVIEW

**Comment by WillAyd** [comment]
+/- 0 on this; I don't know that this is necessarily clearer as a index method but no major objections

**Comment by jbrockmendel** [comment]
@WillAyd gentle ping

**Comment by WillAyd** [comment]
I don't have a strong enough preference on this to go either way. @pandas-dev/pandas-core for other thoughts if a blocker

**Comment by jreback** [comment]
will have a look soon

**Comment by jbrockmendel** [comment]
this isnt a big deal, just a code smell that id like to get out of the way

**Review by jorisvandenbossche — APPROVED** [review]

**Comment by toobaz** [comment]
+1: ideally ``if isinstance(labels, ABCMultiIndex):`` should never appear outside ``pandas/core/indexes/multi.py``. Even clearer would be to remove the need for (non-multi) ``Index`` to support this method (but I didn't check how complicated it is - and this PR is certainly good per se).

**Comment by jreback** [comment]
thanks, yeah getting Index/MI closer and closer but there will always be ovrrides.

File: pandas/core/indexes/multi.py [hunk 1]
```diff
@@ -2396,6 +2396,35 @@ def _convert_listlike_indexer(self, keyarr):
 
         return indexer, keyarr
 
+    def _get_partial_string_timestamp_match_key(self, key):
+        """
+        Translate any partial string timestamp matches in key, returning the
+        new key.
+
+        Only relevant for MultiIndex.
```
Comment (jreback) [inline pandas/core/indexes/multi.py]: can remove this comment at some point

**Comment by toobaz** [comment]
> thanks, yeah getting Index/MI closer and closer but there will always be ovrrides.

Sure! I was just saying that ideally overrides should be only in MI code.

**Pull request merged**